src/cache.ts

```diff
@@ -0,0 +1,49 @@
+import { Cache } from "@raycast/api";
+import { hash } from "./helpers";
+
+/**
+ * Wraps a function with caching functionality using Raycast's Cache API.
+ * Allows for caching of expensive functions like paginated API calls.
+ *
+ * @param fn - The async function to cache results from
+ * @param options - Optional configuration for the cache behavior
+ * @param options.key - Custom cache key (defaults to stringified function)
+ * @param options.validate - Optional validation function for cached data
+ * @param options.maxAge - Maximum age of cached data in milliseconds
+ * @param options.namespace - Optional namespace for the cache
+ * @returns The result of the function, either from cache or fresh execution
+ *
+ * @example
+ * ```ts
+ * const data = await withCache(
+ *   async () => fetchExpensiveData(),
+ *   { maxAge: 5 * 60 * 1000 } // Cache for 5 minutes
+ * );
+ * ```
+ */
+export async function withCache<T>(
+  fn: () => Promise<T>,
+  options?: { key?: string; validate?: (data: T) => boolean; maxAge?: number; namespace?: string },
+): Promise<T> {
+  const cache = new Cache({ namespace: options?.namespace });
+
+  const key = options?.key ?? hash(fn);
+  const cached = cache.get(key);
+  if (cached) {
+    const { data, timestamp } = JSON.parse(cached);
+    const isExpired = options?.maxAge && Date.now() - timestamp > options.maxAge;
+    if (!isExpired && (!options?.validate || options.validate(data))) {
+      return data;
+    }
+  }
+
+  const result = await fn();
+  cache.set(
+    key,
+    JSON.stringify({
```

Comment:
instead of just `JSON.stringify`, let's use `JSON.stringify(..., replacer)` and `JSON.parse(..., reviver)` (both of which you can find in `helpers.ts`) to hand Date and Buffer

Reply:
Good shout, added!